tests/conftest.py

```diff
@@ -89,3 +94,54 @@ def _snapshot(**kwargs):
             yield snapshot
 
     return _snapshot
+
+
+def run_function_from_file(item):
+    file, _, func = item.location
+    marker = item.get_closest_marker("run")
+    expected_status = marker.kwargs.get("status", 0)
+    expected_out = marker.kwargs.get("out", "").encode("utf-8")
+    expected_err = marker.kwargs.get("err", "").encode("utf-8")
+
+    with open(file, "rb") as f:
+        source_bytes = f.read()
+        t = ast.parse(source_bytes.decode("utf-8"))
+        for node in t.body:
+            if isinstance(node, ast.FunctionDef) and node.name == func:
+                t.body = node.body
+                break
+
+        with NamedTemporaryFile(suffix=".pyc") as fp:
+            fp.write(
+                importlib._bootstrap_external._code_to_hash_pyc(
+                    compile(t, file, "exec"),
+                    importlib.util.source_hash(source_bytes),
+                    False,
+                )
+            )
+            fp.flush()
+
+            out, err, status, pid = call_program(sys.executable, fp.name)  # TODO: get args from marker
+            excinfo = None
+            if status != expected_status:
+                excinfo = AssertionError("Expected status %s, got %s" % (expected_status, status))
+            elif out != expected_out:
+                excinfo = AssertionError("Expected out %s, got %s" % (expected_out, out))
+            elif err != expected_err:
+                excinfo = AssertionError("Expected err %s, got %s" % (expected_err, err))
+
+            return TestReport.from_item_and_call(
+                item, CallInfo(result=None, excinfo=excinfo, start=0.0, stop=0.0, duration=0.0, when="call")
+            )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_runtest_protocol(item):
+    if item.get_closest_marker("run"):
```

Comment:
I'd probably go for a `run_in_subprocess`/`subprocess`/`in_subprocess`, or something a little more descriptive.